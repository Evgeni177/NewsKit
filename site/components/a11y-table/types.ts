import {TableProps} from '../table';

export interface A11yTableProps {
  name: string;
  description: string;
  table: TableProps;
}
