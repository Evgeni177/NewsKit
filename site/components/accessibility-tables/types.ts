export interface AccessibilityTablesProps {
  focusOrder?: {
    name: string;
    description: string;
    table: {
      columns: string[];
      rows: FocusOrderProps[];
    };
  };
  aria?: {
    name: string;
    description: string;
    table: {
      columns: string[];
      rows: AriaProps[];
    };
  };
  interaction?: {
    name: string;
    description: string;
    table: {
      columns: string[];
      rows: InteractionProps[];
    };
  };
}

export interface FocusOrderProps {
  order: string[];
  element: string | JSX.Element;
  role?: string;
}

export interface AriaProps {
  category: string;
  attribute: string;
  value: string;
  description: string;
}

export interface InteractionProps {
  command: string[];
  description: string;
}
