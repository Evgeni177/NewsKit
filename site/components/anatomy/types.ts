import {ImageProps} from '../../../src/image/types';

export interface AnatomyProps {
  media: ImageProps;
  list: Array<string>;
}
