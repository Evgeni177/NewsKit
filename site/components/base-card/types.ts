import {CardProps} from 'newskit';

export interface BaseCardProps extends CardProps {
  title?: string;
  children: React.ReactNode;
}
