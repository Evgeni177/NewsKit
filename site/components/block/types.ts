import React from 'react';
import {
  FlexDirectionProperty,
  JustifyContentProperty,
  AlignItemsProperty,
  DisplayProperty,
  OverflowProperty,
  BorderStyleProperty,
  ColorProperty,
  AlignSelfProperty,
  PositionProperty,
} from 'csstype';
import {ColorKeys, TypePresetKeys, SizingKeys, Responsive} from 'newskit';

export interface BlockProps {
  as?: React.ElementType;
  $color?: Responsive<ColorKeys | ColorProperty>;
  $backgroundColor?: Responsive<ColorKeys | ColorProperty>;
  $font?: Responsive<TypePresetKeys>;
  $border?: Responsive<SizingKeys | string>;
  $borderTop?: Responsive<SizingKeys | string>;
  $borderRight?: Responsive<SizingKeys | string>;
  $borderBottom?: Responsive<SizingKeys | string>;
  $borderLeft?: Responsive<SizingKeys | string>;
  $borderStyle?: Responsive<BorderStyleProperty>;
  $borderWidth?: Responsive<SizingKeys | string>;
  $borderColor?: Responsive<ColorKeys | ColorProperty>;
  $margin?: Responsive<SizingKeys | string>;
  $marginTop?: Responsive<SizingKeys | string>;
  $marginRight?: Responsive<SizingKeys | string>;
  $marginBottom?: Responsive<SizingKeys | string>;
  $marginLeft?: Responsive<SizingKeys | string>;
  $padding?: Responsive<SizingKeys | string>;
  $paddingTop?: Responsive<SizingKeys | string>;
  $paddingRight?: Responsive<SizingKeys | string>;
  $paddingBottom?: Responsive<SizingKeys | string>;
  $paddingLeft?: Responsive<SizingKeys | string>;
  $width?: Responsive<SizingKeys | string>;
  $height?: Responsive<SizingKeys | string>;
  $maxWidth?: Responsive<SizingKeys | string>;
  $maxHeight?: Responsive<SizingKeys | string>;
  $minWidth?: Responsive<SizingKeys | string>;
  $minHeight?: Responsive<SizingKeys | string>;
  $display?: Responsive<DisplayProperty>;
  $flexWrap?: Responsive<boolean>;
  $flexDirection?: Responsive<FlexDirectionProperty>;
  $flexGrow?: Responsive<number>;
  $justifyContent?: Responsive<JustifyContentProperty>;
  $alignItems?: Responsive<AlignItemsProperty>;
  $alignSelf?: Responsive<AlignSelfProperty>;
  $position?: Responsive<PositionProperty>;
  $top?: Responsive<SizingKeys | string>;
  $right?: Responsive<SizingKeys | string>;
  $bottom?: Responsive<SizingKeys | string>;
  $left?: Responsive<SizingKeys | string>;
  $overflow?: OverflowProperty;
}
