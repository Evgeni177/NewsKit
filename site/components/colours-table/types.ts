import {TableRow} from '../table';

export interface ComponentColoursTableProps {
  components: ColoursTableProps[];
}

export interface ColoursTableProps {
  title: string;
  summary: string;
  description?: string;
  tabs: {
    header: string;
    rows: TableRow[];
    description: string;
  }[];
}
