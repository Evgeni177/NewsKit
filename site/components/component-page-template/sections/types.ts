import {SectionIntroductionProps} from '../../section-introduction';
import {ComponentAPIProps} from '../../component-api';
import {MetaProps} from '../../meta/types';
import {ComplianceProps} from '../../compliance';
import {PlaygroundProps} from '../../playground/types';
import {AnatomyProps} from '../../anatomy';
import {MediaListProps} from '../../media-list';
import {PageIntroductionProps} from '../../page-introduction';
import {FeatureCardProps} from '../../feature-card';

type IntroductionText = {introduction: string};

export interface IntroductionSectionProps {
  pageIntroduction: PageIntroductionProps;
  meta: MetaProps;
}
export interface InteractiveDemoSectionProps {
  introduction: string;
  playground: PlaygroundProps;
}

export interface AccessibilityTablesProps {
  focusOrder?: {
    title: string;
    description: string;
    table: {
      columns: string[];
      rows: {
        order: string[];
        element: string | JSX.Element;
        role?: string;
      }[];
    };
  };
  aria?: {
    title: string;
    description: string;
    table: {
      columns: string[];
      rows: {
        category: string;
        attribute: string;
        value: string;
        description: string | JSX.Element;
      }[];
    };
  };
  interaction?: {
    title: string;
    description: string;
    table: {
      columns: string[];
      rows: {
        command: string[];
        description: string | JSX.Element;
      }[];
    };
  };
}

export type AnatomySectionProps = AnatomyProps & IntroductionText;
export type OptionsSectionProps = MediaListProps & IntroductionText;
export type StatesSectionProps = MediaListProps & IntroductionText;
export type BehaviorsSectionProps = MediaListProps & IntroductionText;
export type UsageSectionProps = MediaListProps & IntroductionText;
export type AccessibilitySectionProps = AccessibilityTablesProps &
  IntroductionText;
export type SEOSectionProps = Omit<SectionIntroductionProps, 'children'> &
  IntroductionText;
export type ComponentAPISectionProps = ComponentAPIProps & IntroductionText;
export type ComplianceSectionProps = ComplianceProps;
export type RelatedComponentsSectionProps = MediaListProps & IntroductionText;
export type OnwardJourneySectionProps = FeatureCardProps;
