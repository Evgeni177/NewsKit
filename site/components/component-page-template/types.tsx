import {SectionIntroductionProps} from '../section-introduction';
import {AccessibilityTablesProps} from '../accessibility-tables';
import {MetaProps} from '../meta/types';
import {LayoutProps} from '../layout';
import {ComplianceProps} from '../compliance';
import {UsageProps} from '../usage';
import {PlaygroundProps} from '../playground/types';
import {AnatomyProps} from '../anatomy';
import {MediaListProps} from '../media-list';
import {PageIntroductionProps} from '../page-introduction';
import {FeatureCardProps} from '../feature-card';

export type PropTableProps = {
  name: string;
  type: string;
  description: string | JSX.Element;
  default?: string;
  required?: boolean;
};

export interface PropsProps {
  summary: React.ReactNode;
  columns: string[];
  rows: PropTableProps[];
}

export interface ComponentPageTemplateProps {
  layoutProps: LayoutProps;
  componentDefaultsKey: string;
  pageIntroduction: PageIntroductionProps;
  meta: MetaProps;
  interactiveDemo?: {
    introduction: string;
    playground: PlaygroundProps;
  };
  anatomy?: AnatomyProps & {introduction: string};
  variations?: MediaListProps & {introduction: string};
  behaviors?: MediaListProps & {introduction: string};
  usage?: UsageProps & {introduction: string};
  accessibility?: AccessibilityTablesProps & {introduction: string};
  seo?: SectionIntroductionProps & {introduction: string};
  props?: PropsProps;
  overrides?: PropsProps;
  compliance?: ComplianceProps;
  related?: MediaListProps & {introduction: string};
  featureCard?: FeatureCardProps;
}
