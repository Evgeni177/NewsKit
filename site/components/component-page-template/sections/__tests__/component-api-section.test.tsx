import React from 'react';
import {ComponentAPISection} from '..';
import {renderToFragmentWithTheme} from '../../../../utils/test-utils';
import {ComponentAPISectionProps} from '../types';

jest.mock(
  'newskit',
  require('../../../../utils/test-utils').mockNewsKitComponents(
    'Grid',
    'Cell',
    'Tabs',
    'Tab',
  ),
);

jest.mock('../../../table/table');

describe('ComponentAPISection', () => {
  test('renders section as expected', () => {
    const props: ComponentAPISectionProps = {
      introduction: 'component api introduction text goes here',
      components: [
        {
          title: 'Component',
          summary: 'Summary for the component variant',
          propsTable: {
            columns: ['Name', 'Type', 'Default', 'Description', 'Required'],
            rows: [
              {
                name: 'children',
                type: 'string',
                required: true,
                description: (
                  <>The content of the Component is passed as the children.</>
                ),
              },
            ],
          },
          overridesTable: {
            columns: ['Name', 'Type', 'Description'],
            rows: [
              {
                name: 'spaceInset',
                type: 'MQ<string>',
                description:
                  'If provided, this overrides the padding of the Component.',
              },
            ],
          },
        },
      ],
    };

    const fragment = renderToFragmentWithTheme(ComponentAPISection, props);
    expect(fragment).toMatchSnapshot();
  });
});
