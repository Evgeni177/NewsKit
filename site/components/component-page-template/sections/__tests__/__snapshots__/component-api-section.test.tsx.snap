// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentAPISection renders section as expected 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin-top: -24px;
  padding-top: 24px;
}

.emotion-1 {
  margin-bottom: 24px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding: 0;
  background-color: transparent;
}

.emotion-4 {
  padding: 1px 0px;
}

@media screen and (max-width: 767px) {
  .emotion-4 {
    font-family: 'Poppins',sans-serif;
    font-size: 28px;
    line-height: 1.2857142857142858;
    font-weight: 700;
    letter-spacing: 0;
  }

  .emotion-4::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: -0.18348214285714293em;
  }

  .emotion-4::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: -0.4334821428571429em;
  }
}

@media screen and (min-width: 768px) {
  .emotion-4 {
    font-family: 'Poppins',sans-serif;
    font-size: 40px;
    line-height: 1.2;
    font-weight: 700;
    letter-spacing: 0;
  }

  .emotion-4::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: -0.140625em;
  }

  .emotion-4::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: -0.390625em;
  }
}

.emotion-5 {
  display: inline;
  font: inherit;
  margin: 0;
  color: #162740;
}

.emotion-5 svg {
  fill: #162740;
}

.emotion-6 {
  margin-bottom: 40px;
}

.emotion-7 {
  margin: 0;
  padding: 1px 0;
  color: #555E74;
  font-family: 'Poppins',sans-serif;
  font-size: 20px;
  line-height: 1.6;
  font-weight: 500;
  letter-spacing: 0;
}

.emotion-7 svg {
  fill: #555E74;
}

.emotion-7::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.34062500000000007em;
}

.emotion-7::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.5906250000000001em;
}

@media screen and (max-width: 767px) {
  .emotion-9 {
    margin-bottom: 16px;
  }
}

@media screen and (min-width: 768px) {
  .emotion-9 {
    margin-bottom: 20px;
  }
}

.emotion-10 {
  margin: 0;
  padding: 1px 0;
  color: #162740;
  display: inline-block;
}

.emotion-10 svg {
  fill: #162740;
}

@media screen and (max-width: 1439px) {
  .emotion-10 {
    font-family: 'Poppins',sans-serif;
    font-size: 20px;
    line-height: 1.4;
    font-weight: 700;
    letter-spacing: 0;
  }

  .emotion-10::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: -0.24062499999999998em;
  }

  .emotion-10::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: -0.490625em;
  }
}

@media screen and (min-width: 1440px) {
  .emotion-10 {
    font-family: 'Poppins',sans-serif;
    font-size: 24px;
    line-height: 1.3333333333333333;
    font-weight: 700;
    letter-spacing: 0;
  }

  .emotion-10::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: -0.20729166666666665em;
  }

  .emotion-10::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: -0.45729166666666665em;
  }
}

.emotion-11 {
  margin-bottom: 32px;
}

.emotion-12 {
  margin: 0;
  padding: 1px 0;
  color: #555E74;
}

.emotion-12 svg {
  fill: #555E74;
}

@media screen and (max-width: 1439px) {
  .emotion-12 {
    font-family: 'DM Sans',sans-serif;
    font-size: 16px;
    line-height: 1.5;
    font-weight: 400;
    letter-spacing: 0;
  }

  .emotion-12::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: -0.415625em;
  }

  .emotion-12::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: -0.446875em;
  }
}

@media screen and (min-width: 1440px) {
  .emotion-12 {
    font-family: 'DM Sans',sans-serif;
    font-size: 18px;
    line-height: 1.3333333333333333;
    font-weight: 400;
    letter-spacing: 0;
  }

  .emotion-12::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: -0.33229166666666665em;
  }

  .emotion-12::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: -0.36354166666666665em;
  }
}

<div
    data-comp="NewsKit Cell"
    data-props="{
  \\"xs\\": 12
}"
    data-testid="Cell"
  >
    <section
      class="emotion-0"
      data-toc-indexed="API"
      id="component-api"
    >
      <div
        data-comp="NewsKit Grid"
        data-props="{
  \\"lgMargin\\": \\"space000\\",
  \\"xsRowGutter\\": \\"space000\\"
}"
        data-testid="Grid"
      >
        <div
          data-comp="NewsKit Cell"
          data-props="{
  \\"xs\\": 12,
  \\"md\\": 8,
  \\"lg\\": 6,
  \\"mdOffset\\": 1
}"
          data-testid="Cell"
        >
          <div
            class="emotion-1"
          >
            <div
              class="emotion-2"
            >
              <div
                class="emotion-3"
              >
                <section
                  class="emotion-4"
                >
                  <h2
                    class="nk-headline-heading emotion-5"
                  >
                    API
                  </h2>
                </section>
              </div>
            </div>
          </div>
          <div
            class="emotion-6"
          >
            <p
              class="emotion-7"
            >
              component api introduction text goes here
            </p>
          </div>
        </div>
        <div
          data-comp="NewsKit Cell"
          data-props="{
  \\"xs\\": 12,
  \\"md\\": 10,
  \\"lg\\": 8,
  \\"mdOffset\\": 1
}"
          data-testid="Cell"
        />
        <div
          data-comp="NewsKit Cell"
          data-props="{
  \\"xs\\": 12,
  \\"lg\\": 8,
  \\"mdOffset\\": 1
}"
          data-testid="Cell"
        >
          <div
            class="emotion-6"
          >
            <div
              class="emotion-9"
            >
              <span
                class="emotion-10"
              >
                Component
              </span>
            </div>
            <div
              class="emotion-11"
            >
              <p
                class="emotion-12"
              >
                Summary for the component variant
              </p>
            </div>
            <div
              data-comp="NewsKit Tabs"
              data-props="{
  \\"size\\": \\"small\\"
}"
              data-testid="Tabs"
            >
              <div
                data-comp="NewsKit Tab"
                data-props="{
  \\"label\\": \\"Props\\"
}"
                data-testid="Tab"
              >
                <div>
                  Table Component with props:{
  "columns": [
    "Name",
    "Type",
    "Default",
    "Description",
    "Required"
  ],
  "rows": [
    {
      "name": "children",
      "type": "string",
      "required": true,
      "description": {
        "key": null,
        "ref": null,
        "props": {
          "children": "The content of the Component is passed as the children."
        },
        "_owner": null,
        "_store": {}
      }
    }
  ]
}
                </div>
              </div>
              <div
                data-comp="NewsKit Tab"
                data-props="{
  \\"label\\": \\"Overrides\\"
}"
                data-testid="Tab"
              >
                <div>
                  Table Component with props:{
  "columns": [
    "Name",
    "Type",
    "Description"
  ],
  "rows": [
    {
      "name": "spaceInset",
      "type": "MQ&lt;string&gt;",
      "description": "If provided, this overrides the padding of the Component."
    }
  ]
}
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
  @media screen and (max-width: 767px) {
  .emotion-0 {
    margin-top: 40px;
  }
}

@media screen and (min-width: 768px) {
  .emotion-0 {
    margin-top: 80px;
  }
}

@media screen and (max-width: 767px) {
  .emotion-0 {
    margin-bottom: 40px;
  }
}

@media screen and (min-width: 768px) {
  .emotion-0 {
    margin-bottom: 80px;
  }
}

.emotion-1 {
  border-style: solid;
  border-color: #E4E4E4;
  border-width: 1px;
  border-width: 0px;
  margin: 0;
  border-top-width: 1px;
  width: 100%;
}

<div
    data-comp="NewsKit Cell"
    data-props="{
  \\"xs\\": 12,
  \\"md\\": 10,
  \\"lg\\": 8,
  \\"mdOffset\\": 1
}"
    data-testid="Cell"
  >
    <div
      class="emotion-0"
    >
      <hr
        aria-hidden="true"
        class="emotion-1"
        data-testid="divider"
      />
    </div>
  </div>
</DocumentFragment>
`;
